### PR TITLE
feat(config): support byte suffixes

### LINKED ISSUE #572: Implement byte config suffixes
Just as with #545, the config options used to configure byte sizes should be written with a suffix so its clear if the value is to be interpreted as bytes, kilobytes, megabytes or gigabytes.

Here's the options that are durations:

Config option | Release | Default
--------------|---------|---------
`apiRequestSize` | 2.x | 750kb

We need to decide what should happen in 2.x when a suffix isn't given:

1. ~~Should we ignore the given config value, output a warning, and fall back to the default value?~~
1. ~~Should we output an error and make the agent inactive?~~
1. ~~Should we throw an error?~~
1. Assume the user meant to use the unit that was the default in 1.x, but output a deprecation warning that it will be required in v3.0.0

### PR DESCRIPTION
**Pull request by Qard**
Fixes #572

### Checklist

- [x] Implement code
- [x] Add tests
- [x] Update documentation

### CODE REVIEW

File: docs/agent-api.asciidoc [hunk 1]
```diff
@@ -560,8 +561,8 @@ Use the <<api-request-size,`apiRequestSize`>> config option to control the byte
 [[api-request-size]]
 ===== `apiRequestSize`
 
-* *Type:* Number
-* *Default:* `1048576` (1 MiB)
+* *Type:* String
+* *Default:* `1mb` (1 MiB)
```
Comment (watson) [inline docs/agent-api.asciidoc]: No reason to write the MB size in brackets anymore

Comment (Qard) [reply docs/agent-api.asciidoc]: fixed

**Comment by Qard** [comment]
I've added docs for the valid suffixes. Technically `tb` is also valid, with the use of the `bytes` module, but is not included in our spec. Is it fine for it to be supported and just not documented, or should we do another layer of regex validation like I did for times?

**Comment by watson** [comment]
This wasn't clear in the original issue, but the parsing rules should actually be quite strict:

> - we accept values matching the regular expression `^(\d+)(B|KB|MB|GB)$` (with case insensitivity enabled). The multipliers are base-2.
> - parsing will be strict, to ensure compatibility between agents: no white space, no additional suffixes, no floating point values

The `bytes` module is "unfortunately" cleaver and will parse both whitespace and floats. While this is nice for Node.js users, it can introduce incompatibilities with the other agents if config values is shared.

**Comment by Qard** [comment]
Switched to a regex parser, like I did for the time suffix parser.

**Pull request merged**